community_tasks/arabic_evals.py

```diff
@@ -832,6 +835,162 @@ def __init__(
 ]
 
 
+class JudgeMetricWrapper(Metric):  # Extend from Metric
+    def __init__(self, judge: JudgeLM):
+        self.judge = judge
+        self.metric_name = "llm_as_judge"  # Define a metric name
+        self.category = MetricCategory.LLM_AS_JUDGE  # Add the category attribute
+        self.corpus_level_fn = self.aggregate_scores  # Define the corpus level function
+        self.sample_level_fn = self._sample_level_fn
+        self.higher_is_better = (True,)
+        self.use_case = MetricUseCase.NONE
+
+    def compute(self, responses: list[str], formatted_docs: list[Doc], **kwargs) -> dict[str, float]:
+        """
+        Compute the score using the judge's evaluate_answer method.
+
+        Args:
+            predictions (list[str]): The predicted answers.
+            formatted_docs (list[Doc]): The formatted documents containing questions and gold answers.
+
+        Returns:
+            dict[str, float]: A dictionary containing the evaluation scores.
+        """
+        results = []
+        for i, doc in enumerate(formatted_docs):
+            question = doc.query
+            gold = doc.choices[doc.gold_index] if doc.gold_index is not None else None
+            answer = responses[i][0].result[0]
+
+            # Directly use judge.evaluate_answer here
+            score, _, _ = self.judge.evaluate_answer(question, answer, options=None, gold=gold)
+
+            results.append({self.metric_name: score})
+        # Return a dictionary with a key that can be accessed
+        return results
+
+    def aggregate_scores(self, scores: list[dict]) -> float:
+        """Aggregate scores from the compute method."""
+        return sum(scores) / len(scores) if scores else 0.0
+
+    def _sample_level_fn(self):
+        return None
+
+
+def qa_prompt_arabic(line: Dict, task_name: str = None) -> Doc:
+    """Format the prompt for question answering with candidates"""
+
+    # Check the input line structure
+
+    question = str(line["question"])
+
+    # Convert candidates to string if it isn't already
+    if isinstance(line["candidates"], list):
+        candidates = [str(c) for c in line["candidates"]]
+    else:
+        candidates = str(line["candidates"]).split("\n")
+
+    # Clean up candidates
+    candidates = [c.strip() for c in candidates if c.strip()]
+
+    instruction = "بناءً على السياقات المقترحة التالية، اجب عن السؤال التالي"
+    query = f"{instruction}\n\nالسؤال:\n{question}\n\nالسياقات المقترحة:\n{', '.join(candidates)}\n"
+
+    # Ensure gold_answer is a string
+    gold_answer = str(line.get("gold_answer", ""))  # Ensure this is set correctly
+
+    # Create Doc with proper string types
+    doc = Doc(
+        task_name=task_name or "alrage",
+        query=query,
+        instruction=instruction,
+        choices=[gold_answer],  # Ensure this is populated correctly
```

Comment:
"# Ensure this is populated correctly" -> do you mean adding the gold_answer in choices ensures this?